### ISSUE TITLE
WISH: Add (optional) support for MuTect2

### ISSUE THREAD

**Issue by HenrikBengtsson** [UCSF-Costello-Lab/LG3_Pipeline#115]
If doable, we should look into adding support for using [MuTect2](https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.php) instead of MuTect1.

**Comment by ivan108** [comment]
OK, I am on it! I assume you mean the latest MuTect2 which comes with GATK 4.0 (currently 4.0.12.0)?

**Comment by HenrikBengtsson** [comment]
Yes - this request came from the Ziv lab (but I'd assume others may ask for this as well).

BTW, will this cause issues of the existing downstream pipeline steps?  I remember you said you ran into those issues when you rewrote (your version of) the pipeline a while ago.

**Comment by ivan108** [comment]
Yes, I see 2 parts of the task: 1. implement somatic mutations calling based on MuTect2 (I already made some progress there), and 2. somehow to integrate the  std. MuTect2 output, which is just a vcf file, into custom final report, which includes custom annotations, some filtering and custom QC...

**Comment by ivan108** [comment]
I created a simplified version of GATK4/MuTect2 sub-pipeline, using Broad reference implementation as a guide:
https://github.com/broadinstitute/gatk/blob/master/scripts/mutect2_wdl/mutect2.wdl
So far I used GATK4 modules (scripts/Mutect2_TvsN.sh):
- GetSampleName (beta)
- MuTect2
- FilterMutectCalls (basic filtering of raw calls)
- Funcotator (functional annotations of filtered calls)

In the end we get filtered and annotated VCF file
